pkg/insights/resyncer.go

```diff
@@ -361,7 +361,10 @@ func (r *resyncer) createOrUpdateMeshInsight(mesh string) error {
 
 	for _, dpOverview := range dpOverviews.Items {
 		dpInsight := dpOverview.Spec.DataplaneInsight
-		dpSubscription, _ := dpInsight.GetLatestSubscription()
+		dpSubscription, ok := dpInsight.GetLastSubscription().(*mesh_proto.DiscoverySubscription)
+		if !ok {
```

Comment:
@lobkovilya I added this type of inference check here because one of the valid test started failing without this check.

can we add this to the other places or is there any better way to handle this?

Reply:
what test is failing without the type check?

Reply:
1. should return gateway in services
2. should count data planes by mTLS backends

Reply:
Oh I see, actually we should return not just `nil` but `(*DiscoverySubscription)(nil)` so it'd be possible to cast:
```go
func (x *DataplaneInsight) GetLastSubscription() generic.Subscription {
	if len(x.GetSubscriptions()) == 0 {
		return (*DiscoverySubscription)(nil)
	}
	return x.GetSubscriptions()[len(x.GetSubscriptions())-1]
}
```
In this case tests should pass without the check

Reply:
added the casting to other resources `GetLastSubscription` also.